rust/operator-binary/src/storage_quantity.rs

```diff
@@ -0,0 +1,54 @@
+use std::{num::ParseFloatError, ops::Mul};
+
+use snafu::{ResultExt, Snafu};
+use stackable_operator::k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+
+#[derive(Debug, Snafu)]
+#[snafu(module)]
+pub enum FromK8sError {
+    #[snafu(display("unknown unit {unit:?}"))]
+    UnknownUnit { unit: String },
+    #[snafu(display("failed to parse amount {amount:?}"))]
+    UnparseableAmount {
+        amount: String,
+        source: ParseFloatError,
+    },
+}
+
+pub struct StorageQuantity {
+    mebibytes: f64,
+}
+
+impl StorageQuantity {
```

Comment:
Some unit tests for this would be great.
Can't we use the already existing helpers in operator-rs? Don't we already have Code that parse a Memory Quantitiy and convert them to a floating point in megabytes? `to_java_heap_value(heap_size_definition, 0.8, BinaryMultiple::Mebi)` or something like that?

Reply:
Sort of. We have `Memory`, but its underlying machinery is not exposed publicly, only the JVM-memory-specific parts.

We could expose that and keep the Nifi-specific conversion stuff in nifi-op.

Reply:
Ok for me. Moving it to operator-rs would be great but i wouldn't block this bugfix on that as we want to include it in 22.11

Reply:
Done